apps/src/pythonlab/pyodideWorkerManager.ts

```diff
@@ -28,12 +33,22 @@ const setUpPyodideWorker = () => {
       case 'syserr':
         // We currently treat sysout and syserr the same, but we may want to
         // change this in the future. Test output goes to syserr by default.
-        if (message.startsWith(MATPLOTLIB_IMG_TAG)) {
+        if (message.startsWith(MessageTag.MATPLOTLIB_IMG)) {
           // This is a matplotlib image, so we need to append it to the output
-          const image = message.slice(MATPLOTLIB_IMG_TAG.length + 1);
+          const image = message.slice(MessageTag.MATPLOTLIB_IMG.length + 1);
           consoleManager?.writeImage(image);
           break;
         }
+        if (message.startsWith(MessageTag.NEIGHBORHOOD_SIGNAL)) {
+          const neighborhood =
+            CodebridgeRegistry.getInstance().getNeighborhood();
+          if (neighborhood) {
+            // Parse message string to NeighborhoodSignal.
+            const data = parseMessageToNeighborhoodSignal(message);
+            neighborhood.handleSignal(data as NeighborhoodSignal);
```

Comment:
Do we need to cast this? It looks like `parseMessageToNeighborhoodSignal` returns a `NeighborhoodSignal`

Reply:
Nope! I think I had to cast before I had created a helper function. Updated!